### PR TITLE
temporary index will now add docs on main thread and will have no GC

### PR DESCRIPTION
**Pull request by MeirShpilraien**

### CODE REVIEW

File: src/document.c [hunk 1]
```diff
@@ -129,7 +129,7 @@ RSAddDocumentCtx *NewAddDocumentCtx(IndexSpec *sp, Document *b, const char **err
   aCtx->client.bc = NULL;
   aCtx->next = NULL;
   aCtx->specFlags = sp->flags;
-  aCtx->indexer = GetDocumentIndexer(sp->name);
+  aCtx->indexer = GetDocumentIndexer(sp->name, sp->timeout == -1);
```
Comment (mnunberg) [inline src/document.c]: Please set something like `sp->temporary`.

File: src/indexer.c [hunk 2]
```diff
@@ -545,7 +545,8 @@ static DocumentIndexer *findDocumentIndexer(const char *specname) {
 
 // Creates a new DocumentIndexer. This initializes the structure and starts the
 // thread. This does not insert it into the list of threads, though
-static DocumentIndexer *NewDocumentIndexer(const char *name) {
+// todo: remove the withIndexThread var once we switch to threadpool
+static DocumentIndexer *NewDocumentIndexer(const char *name, bool withIndexThread) {
```
Comment (mnunberg) [inline src/indexer.c]: `(const char *name, int options)`...
then:

```
#define INDEXER_THREADLESS 0x01
```

and finally

```
NewDocumentIndexer(name, INDEXER_THREADLESS)
```

File: src/document_add.c [hunk 3]
```diff
@@ -161,6 +161,10 @@ static int doAddDocument(RedisModuleCtx *ctx, RedisModuleString **argv, int argc
     RedisModule_ReplyWithError(ctx, "Unknown index name");
     goto cleanup;
   }
+  if (canBlock) {
```
Comment (mnunberg) [inline src/document_add.c]: Move lines 155 here, and then:

`canBlock = !sp->temporary && CheckConcurrentSupprt(...)`

File: src/spec.c [hunk 4]
```diff
@@ -604,6 +595,34 @@ void IndexSpec_Free(void *ctx) {
...
+  RedisModule_ThreadSafeContextUnlock(threadCtx);
+  RedisModule_FreeThreadSafeContext(threadCtx);
+  return NULL;
+}
+
+void IndexSpec_Free(void *ctx) {
+
+  IndexSpec *spec = ctx;
+
+  if (spec->timeout != -1) {
```
Comment (mnunberg) [inline src/spec.c]: ->temporary... maybe other flags for this? maybe we should use a function to hide?

**Comment by MeirShpilraien** [comment]
@mnunberg I added another flag to spec flags field called Index_Temporary, pls take another look.

**Pull request merged**